server/server.go

```diff
@@ -152,18 +167,28 @@ func (server appointmentsServer) GetAppointments(ctx context.Context,
 		return nil, status.Error(codes.PermissionDenied, permissionDeniedMessage)
 	}
 
-	// Assuming req.GetDate() returns a string in "2006-01-02" format
-	dateStr := req.GetDate()
-	date, err := time.Parse("2006-01-02", dateStr)
-	if err != nil {
-		return nil, status.Error(codes.InvalidArgument, fmt.Errorf("failed to parse date: %w", err).Error())
+	if req.GetSkip() < 0 {
+		return nil, status.Error(codes.InvalidArgument, "skip has to be a non-negative integer")
+	}
+	if req.GetLimit() <= 0 {
+		return nil, status.Error(codes.InvalidArgument, "limit has to be a positive integer")
+	}
+	if req.GetLimit() > maxPaginationLimit {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("maximum allowed limit values is %d", maxPaginationLimit))
 	}
 
 	// Fetch appointments based on filters
 	baseQuery := server.db.NewSelect().Model((*Appointment)(nil))
 
 	// Filter by date range
-	baseQuery = baseQuery.Where("start_time >= ?", date).Where("end_time <= ?", date.AddDate(0, 0, 1))
+	dateStr := req.GetDate()
+	if dateStr != "" {
+		date, dateErr := time.Parse("2006-01-02", dateStr)
```

Comment:
Make date format a const.

I recall seeing this somewhere else, note that if different components need to be aligned on this point, it is best to extract the formatting (and other consts) to a new data-types repo or whatever meaningful name you select. Alternatively, if they should be aligned, then the minimum would be for one to import it from the other.

Reply:
I added the relevant const, where should I open an issue regarding the data-types repo?

Reply:
@nativraz1 backlog

Reply:
done. (https://github.com/TekClinic/backlog/issues/36#issue-2227531840)